x/ibc/transfer/keeper/msg_server.go

```diff
@@ -68,12 +60,10 @@ func (k Keeper) Transfer(goCtx context.Context, msg *types.MsgTransfer) (*types.
 			)
 		}()
 
-		// update the denom and proceed with regular transfer
-		msg.Token.Denom = tokenPair.Denom
```

Comment:
Don't we need to update the msg?

Reply:
After the last changes  the contract address will not be accepted as a valid denomination. so the ibc transfer msg should already use the coin denom. So there's no need to update